messagequeue/messagequeue.go

```diff
@@ -87,8 +89,16 @@ func (mq *MessageQueue) Startup(ctx context.Context, initialEntries []*wantlist.
 			fullwantlist.AddEntry(e.Cid, e.Priority)
 		}
 		mq.out = fullwantlist
-		mq.work <- struct{}{}
 	}
+
+	if len(initialEntries) > 0 || mq.addEntries(entries, ses) {
+		select {
+		case <-ctx.Done():
+			return
+		case mq.work <- struct{}{}:
```

Comment:
I believe this should be:

```go
select {
case mq.work <- struct{}{}:
default:
}
```

(it's a flag).

Reply:
should be resolved with latest code.